### ISSUE TITLE
[ISSUE] Provider bug(?) - attempting to instantiate UserInstanceProfile resource results in `Endpoint not supported` error

### ISSUE THREAD

**Issue by cprice404-aws** [databrickslabs/terraform-provider-databricks#609]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Databricks Provider issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version

I am attempting to use the provider via Pulumi version 2.24.1.  I recognize that this is unsupported, I would just like to know if there are any known issues that may cause `Endpoint not supported` errors.

### Affected Resource(s)

- databricks_user_instance_profile

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Environment variable names

```
AWS_PROFILE
AWS_REGION
DATABRICKS_HOST
DATABRICKS_TOKEN
```

### Terraform Configuration Files

N/A.  This is the pulumi code that I am trying to use:

```
    const notebook = new databricks.Notebook('deleteme', {
        contentBase64: Buffer.from('foo').toString('base64'),
        language: 'SCALA',
        path: '/deleteme',
    });

    const instanceProfile = new databricks.UserInstanceProfile('databricks-user-instance-profile', {
        userId: 'foo@bar.com',
        instanceProfileId: databricksIamInstanceProfile.arn,
    });
```

Please note that the `Notebook` object creates without issue, but the UserInstanceProfile 

### Debug Output

Pulumi does support the TF_LOG flags and I could provide logs, but when I run with the flags I get about 600k
of output.  That seems excessive but I will provide it upon request.

Here is a gist of the few lines that seem most relevant:

https://gist.github.com/cprice404-aws/7f04bf0fdef1a00b04a2f2ed14fc2adc

### Panic Output

No panic.

### Expected Behavior

The provider should have created a notebook and a UserInstanceProfile.

### Actual Behavior

The provider created a notebook but returned an error `Endpoint not supported.` for the UserInstanceProfile.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:

Probably difficult to reproduce since we are using pulumi, so my inquiry is just about whether or not there are any known issues that would cause `Endpoint not supported`.

### Important Factoids
Are there anything atypical about your accounts that we should know?

**Comment by nfx** [comment]
@cprice404-aws this is not quite terraform debug log i'm seeing. can you check debug output of `terraform-provider-databricks` binary? that one should contain the API call sequence. 

Why do you need to assign an instance profile to a user and not a group? our telemetry doesn't show enough usage for this resource ...

**Comment by nfx** [comment]
and can you call normal terraform instead?...

**Comment by cprice404-aws** [comment]
Thanks for your response.  I understand that my use case is unusual so it's no problem if you're not able to help, I just wanted to see if there were known circumstances that would result in that `Endpoint not supported` message.

Running via pulumi it uses a bridge that wraps the terraform provider so I don't have the normal binary around, nor the logs you're interested in.  I may try to set up a terraform environment so that I can run directly that way, just as a debugging tool, but my environment is pulumi so that won't really be a long-term solution.  If I do set that up then I will share any logs if I am able to repro the failure there.

I could try a group instance profile, I don't have any particular reason for using a user one vs. a group one.  If I try that and get any different behavior I'll also update the issue.

Generally speaking though, is `Endpoint not supported` a known error, and if so, what sorts of things can cause it?

**Comment by cprice404-aws** [comment]
I tried with a `GroupInstanceProfile` and in that case, it says "Group with id foo not found", though there is definitely a group by that name.  It's interesting that it does not trigger the same `Endpoint not supported` error, though.

**Comment by cprice404-aws** [comment]
FYI, I started trying to play around with this via `curl` to make sure I knew what I was trying to call, and realized that the public API docs only mention an `instance-profiles` endpoint.  They don't mention a REST API for "user" or "group" instance profiles.  Then I realized that the terraform provider also exposes an `databricks_instance_profile` resource - the mapping between that and the REST API seems more obvious than the user/group ones.  I am testing that out in my pulumi code and it seems to be working better... I haven't quite got it working yet because it's failing the profile validation, because my profile doesn't have the right AWS policy stuff yet... but it seems like it's getting a lot farther than the other two.
